### ISSUE TITLE
Seq __format__  should return a string

### ISSUE THREAD

**Issue by mdehoon** [biopython/biopython#2332]
### Setup

I am reporting a problem with Biopython version 1.75, Python version 3.6, and operating
system as follows:

CPython
Darwin-18.7.0-x86_64-i386-64bit
1.76.dev0

```python
>>> from Bio.Seq import Seq
>>> from Bio.SeqRecord import SeqRecord
>>> s = Seq("ACGT")
>>> record = SeqRecord(s)
>>> format(record, 'xdna')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __format__ must return a str, not bytes
```

This is coming from the `__format__` method in `SeqRecord`:
```python
        from Bio import SeqIO

        # Easy case, can call string-building function directly
        if format_spec in SeqIO._FormatToString:
            return SeqIO._FormatToString[format_spec](self)

        # Harder case, make a temp handle instead
        if format_spec in SeqIO._BinaryFormats:
            # Return bytes on Python 3
            from io import BytesIO
            handle = BytesIO()
        else:
            from Bio._py3k import StringIO
            handle = StringIO()
        SeqIO.write(self, handle, format_spec)
        return handle.getvalue()
```

i.e. if `format_spec` is in `SeqIO._BinaryFormats`, then on python3 we will be returning bytes.

Something along these lines  may work better:
```python
        from Bio import SeqIO

        # Easy case, can call string-building function directly
        if format_spec in SeqIO._FormatToString:
            return SeqIO._FormatToString[format_spec](self)

        # Harder case, make a temp handle instead
        from Bio._py3k import StringIO
        handle = StringIO()
        try:
            SeqIO.write(self, handle, format_spec)
        except TypeError:
            raise ValueError("cannot format as a string")
        return handle.getvalue()
```

i.e. try to make a string, and give a meaningful error message if it fails.

**Comment by peterjc** [comment]
If Python 3 now insists that ``__format__`` returns a (unicode) string, then for binary formats it should give a meaningful error instead. Do you have older versions of Python already installed to see if and when that changed?

**Comment by mdehoon** [comment]
@peterjc  Python 3.3 at least; it also complains if `__format__` does not return a string.

**Comment by peterjc** [comment]
I think we should raise a ``ValueError``, but in the short term Python 2 can continue as now but with a deprecation warning.

**Comment by mdehoon** [comment]
Fixed by #2333
